tests/main/writable-areas/task.yaml

```diff
@@ -20,19 +25,7 @@ execute: |
 
     echo "Waiting for data writer service to finish..."
     unit="snap.data-writer.service.service"
-    while true; do
-        code=$(sudo systemctl show -p ExecMainCode $unit | sed 's/.*=\([0-9]\+\)/\1/')
-        # The main code will be 0 until the service is no longer running
-        if [ $code -ne 0 ]; then
-            status=$(sudo systemctl show -p ExecMainStatus $unit | sed 's/.*=\([0-9]\+\)/\1/')
-            if [ $status -ne 0 ]; then
-                echo "Service exited $status"
-                exit 1
-            fi
-            break
-        fi
-        sleep 0.1
-    done
+    wait_for_service $unit inactive
```

Comment:
From the docs: "inactive indicates that it is inactive and the previous run was successful or no previous run has taken place yet." What if this runs before the service has actually started?